File: WorkManager.Server/src/WorkManager/Data/DataAccess/HistoryRepository.java
package WorkManager.Data.DataAccess;

import WorkManager.Data.Models.DataContext;
import WorkManager.Data.Models.EFValuesHistory;
import WorkManager.Data.QueryBuilders.DBCondition;
import WorkManager.Data.QueryBuilders.DBConditionOperator;
import WorkManager.Data.QueryBuilders.QueryBuilder;

import java.lang.reflect.InvocationTargetException;
import java.sql.ResultSet;
import java.sql.SQLException;
import java.util.ArrayList;
import java.util.Date;
import java.util.List;

public class HistoryRepository extends Repository {
    public List<EFValuesHistory> GetValuesHistory(DataContext context, Date from, Date to) throws SQLException, InvocationTargetException, NoSuchMethodException, InstantiationException, IllegalAccessException {
        QueryBuilder queryBuilder = new QueryBuilder().Select().From(context.ValuesHistory).OrderByDescending(EFValuesHistory.ModDate);
        if (from != null)
            queryBuilder.Where(new DBCondition(EFValuesHistory.ModDate, DBConditionOperator.GreaterOrEqual, from));
        if (to != null)
            queryBuilder.Where(new DBCondition(EFValuesHistory.ModDate, DBConditionOperator.LessOrEqual, to));
        return executeQuery(EFValuesHistory.class, context, queryBuilder);
    }
}

File: WorkManager/WorkManager.Data/DataAccess/HistoryRepository.cs
using System;
using System.Collections.Generic;
using System.Linq;
using System.Text;
using System.Threading.Tasks;
using WorkManager.Data.Models;

namespace WorkManager.Data.DataAccess
{
    public class HistoryRepository
    {
        public IEnumerable<EFValuesHistory> GetValuesHistory(DataContext context, DateTime? from, DateTime? to)
        {
            return context.EFValuesHistory
                .Where(x => (!from.HasValue || x.ModDate >= from.Value) && (!to.HasValue || x.ModDate >= to.Value))
                .OrderByDescending(x => x.ModDate)
                .ToList();
        }
    }
}
